File: Maximum Subarray/Maximum Subarray.py
def Maximum_Subarray(arr):
    maxSub = arr[0]
    curSum = 0
    for n in arr:
        if curSum < 0:
            curSum = 0
        curSum += n
        maxSub = max(maxSub, curSum)
    return [maxSub]
print(Maximum_Subarray([-2,1,-3,4,-1,2,1,-5,4]))

File: Maximum Subarray/Maximum Subarray.js
function Maximum_Subarray(arr){
    var maxSub = arr[0]
    var curSub = 0
    for(var i = 1; i < arr.length; i++){
        if(curSub < 0){
            curSub = 0
        }
        curSub += arr[i]
        
        maxSub = Math.max(maxSub, curSub)

    }
    return maxSub
}
console.log(Maximum_Subarray([-2,1,-3,4,-1,2,1,-5,4]))